hermes/src/aggregate.rs

```diff
@@ -179,12 +232,41 @@ pub async fn store_update(state: &State, update: Update) -> Result<()> {
     // we can build the message states
     build_message_states(state, accumulator_messages, wormhole_merkle_state).await?;
 
-    state.update_tx.send(()).await?;
+    // Update the aggregate state
+    let mut aggregate_state = state.aggregate_state.write().await;
+
+    // Check if the update is new or out of order
+    match aggregate_state.latest_completed_slot {
+        Some(latest_completed_slot) => {
+            if slot > latest_completed_slot {
+                aggregate_state.latest_completed_slot.replace(slot);
+                state
+                    .api_update_tx
+                    .send(AggregationEvent::New { slot })
+                    .await?;
+            } else {
+                state
+                    .api_update_tx
+                    .send(AggregationEvent::OutOfOrder { slot })
+                    .await?;
+            }
+        }
+        None => {
+            aggregate_state.latest_completed_slot.replace(slot);
+            state
+                .api_update_tx
+                .send(AggregationEvent::New { slot })
+                .await?;
+        }
+    }
 
-    state
-        .last_completed_update_at
-        .write()
-        .await
+    aggregate_state.latest_completed_slot = match aggregate_state.latest_completed_slot {
+        Some(latest_completed_slot) => Some(latest_completed_slot.max(slot)),
+        None => Some(slot),
+    };
```

Comment:
This is just a map, can do either of these which I think are a little clearer:

```suggestion
    aggregate_state.latest_completed_slot = aggregation_state.latest_completed_slot
        .map(|latest| latest.max(slot));
```

or

```suggestion
    aggregate_state.latest_completed_slot.as_mut().map(|latest| *latest = latest.max(slot));
```